src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -895,6 +895,15 @@ void move(UnixPath source, UnixPath target, CopyOption... options)
             x.rethrowAsIOException(source);
         }
 
+        if (sourceAttrs.isDirectory()) {
+            // ensure source can be moved
+            try {
+                access(source, W_OK);
```

Comment:
One small suggestion here is to move this to L900, after UnixFileAttributes.get, as that will avoid the exception handling and means the examination of the source is in one place.

Reply:
Good idea. So changed in cd644e46fed22a1f8b4932b8ee5b54626134ad18.